sdk/state/update.go

```diff
@@ -92,11 +101,28 @@ func (c *Channel) PaymentTxs(p *Payment) (close, decl *txnbuild.Transaction, err
 	return
 }
 
-func (c *Channel) ConfirmPayment(p *Payment) (payment *Payment, fullySigned bool, err error) {
+func (c *Channel) ConfirmPayment(p Payment) (payment Payment, fullySigned bool, err error) {
+	defer func() {
```

Comment:
I like the simplicity of adding a deferred function call here for state cleanup, but I do think this affects readability. I know you mentioned refactoring this later on, but could you leave a few comment lines explaining why this is here?

Reply:
added a comment here that I think make the intent clearer. Lmk if it is still not clear
https://github.com/stellar/experimental-payment-channels/pull/81/commits/69c1b651a70d8a91162de5c6f943b26ceb82b6ab